CONTRIBUTING.md

```diff
@@ -151,25 +151,60 @@ If changes introduced to `master` branch result in conflicts, it should be merge
+
+### Accepting a Pull Request
+
+Check list:
+
+1. At least two approvals by Core Team members; one approval if the author is a Core Team member. Only approvals based on the most recent code version count (ignoring minor changes like fixing a typo).
+2. There are no outstanding questions nor requested changes in the pull request and associated issues.
+3. Title and description appropriately represent the final state of the change.
+4. Proper labels are applied (usually at least a `topic:` and `kind:` label).
+5. Change is based on a fairly recent commit of the `master` branch. When in doubt, merge `master` and wait for CI.
+6. CI is green.
+
+When these conditions are met, a Core Team member can mark the pull request as accepted by adding it to the current development milestone.
+This signals it is scheduled to be merged soon and gives another chance for final reviews.
+
+The current development milestone is typically the milestone for the next release.
+During the freeze period for the next release, feature enhancements are added to the release milestone after the next.
+Freeze periods are announced on the community forums and usually span two weeks before the scheduled date of a minor release.
+
+### Merge Queue
+
+The current [development milestone](https://github.com/crystal-lang/crystal/milestones) serves as a merge queue. Open pull requests on that milestone
+are eligible for being merged.
+
+Pending pull requests should usually stay in the queue for at least one full business day, allowing other reviewers to take a final look at it.
+This wait time can be extended, for example for big changes or when there was a lot of recent activity in the discussion.
+Urgent bug fixes can skip the line.
+
+If reasonable objection or questsions arise while waiting for merge, the pull request must be removed from the milestone until they are resolved.
+
+It's good practice to have a single maintainer responsible for operating the merge queue.
+
+### Merging
+
+Make sure, the pull request has been on the merge queue for some time (usually 1+ business days) and there has not been any
+more recent discussion that questions the current state of the change.
+
+Now, the pull request can finally be merged. Use squash merge to not pollute the version history of the main branch with
+details of the pull request process. For non-trivial changes, the merge commit should contain a short description.
 
 ### For maintainers with push access
 
 1. Do not directly commit to the `master` branch. Always create a feature branch and pull request.
 2. Feature branches should typically be created in your fork. The main repo should only contain essential branches.
+   * CI changes affecting circle CI only run for branches on the main repo. They should be prefixed `ci/` to trigger a maintenance release.
+   * Long-running feature branches that accept contributions must be pushed to the main repo in order to allow PRs targeting that branch.
```

Comment:
Mixing ordered and unordered lists, also indentation is redundant here.

Reply:
Indentation is on purpose. These are sub-items of 2.

Reply:
FWIW @Sija exactly represented what a stock Markdown parser sees 😂 - the 3 spaces aren't enough to mean anything (say, if this was MkDocs)

Reply:
In GFM it's a sub list: https://github.com/straight-shoota/crystal/blob/feature/contrib-merge-process/CONTRIBUTING.md#for-maintainers-with-push-access

Reply:
Shouldn't they be indented with 2 spaces?

Reply:
No, the indent must be the same as the indent of the parent item, which in this case is 3 (`2. `).